### PR TITLE
Extract students-in-filter component from overview and add to STAR reading / math pages

### PR DESCRIPTION
**Pull request by jonchar**
Opening this to get feedback while I work out what's going on in #375. What's happening is that the number of students being filtered via the `SlicePanel` doesn't match up with the number reported in the visualization.

I couldn't reproduce a large discrepancy when running locally, but added a div to display information about the number of students the `SlicePanel` is selecting. I'm not sure whether it should go in `slice_panels.js` or `star_charts_page.js` depending on if the indicator is relevant in other views. Suggestions welcome how to fit it in style-wise of course :smile: . I assume this number should be the same as what gets displayed in the visualization generated by `StarChartsPage`.

This shows the indicator below the columns with 64 students. Just below the cutoff it reads there are 56 students in the visualization.

slicepanel

### CODE REVIEW

**Comment by kevinrobinson** [comment]
@jonchar Thanks!  This might be helpful for you too: https://github.com/studentinsights/studentinsights/issues/375#issuecomment-222262168

**Comment by jonchar** [comment]
I just finished extracting the students-in-filter parts from the `school_overview_page.js` (see https://github.com/studentinsights/studentinsights/blob/master/app/assets/javascripts/school_overview/school_overview_page.js#L212) into its own component. I named the new component `slice_buttons.js` (any suggestions for a clearer name are welcome). I also removed the count that I had previously added into `slice_panels.js` for this PR (see above).

I also added the newly extracted component to `star_charts_page.js` so we now have a students-in-filter count for the STAR Reading / Math pages. This should address #375 aside from further documenting the filtering edge cases @ianaroot found. Not sure if that warrants a separate issue.

Someone should probably have a look at this before merging just to be sure my React is on par :smile: .

**Comment by jonchar** [comment]
Realized I had missed the code for using the ESC key to clear the filters. Just moved that code to `slice_buttons.js` as well. Thanks @erose !

**Comment by erose** [comment]
Jon is this still a WIP or would you like it to get reviewed? I took away the `needs-review` label after we talked about the ESC key stuff, but reapply as needed.

**Comment by jonchar** [comment]
I'm believe I properly moved the ESC key functionality into `slice_buttons.js` and out of `school_overview_page.js ` & `star_charts_page.js`. The buttons work as expected on both the overview and STAR pages. I'm re-applying the `needs-review` however because someone else should have a look since it's my first attempt at making a new component. Other than that I think it's good to go.

File: app/assets/javascripts/components/slice_buttons.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  var colors = window.shared.colors;
+
+  var SliceButtons = window.shared.SliceButtons = React.createClass({
+    displayName: 'SliceButtons',
+      
+    propTypes: {
+      students: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
+      filters: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
+      filtersHash: React.PropTypes.func.isRequired,
+      activeFiltersIdentifier: React.PropTypes.func.isRequired,
```
Comment (kevinrobinson) [inline app/assets/javascripts/components/slice_buttons.js]: I think `SliceButtons` only needs the values for `filtersHash` and `activeFiltersIdentifier`.  So you can have the call sites evaluate those functions and then pass the plain value into `SliceButtons` (the same way that `students` works).

Like this:
```
createEl(SliceButtons, {
  students: this.filteredStudents(),
  filters: this.state.filters,
  filtersHash: this.filtersHash(),
  activeFiltersIdentifier: this.activeFiltersIdentifier(),
  onClearFilters: this.onClearFilters
})
```

File: app/assets/javascripts/components/star_charts_page.js [hunk 2]
```diff
@@ -76,6 +85,14 @@ $(function() {
       this.setState({ filters: updatedFilters });
     },
 
+    onResetClicked: function(e) {
+      this.clearFilters();
+    },
+
+    //onKeyDown: function(e) {
+    //  if (e.keyCode == 27) this.clearFilters();
+    //},
```
Comment (kevinrobinson) [inline app/assets/javascripts/components/star_charts_page.js]: Leftover comments here.

File: app/assets/javascripts/components/star_charts_page.js [hunk 3]
```diff
@@ -186,7 +210,7 @@ $(function() {
         dom.div({}, dateRangeText),
         dom.div({},
           dom.span({}, 'Students: ', props.students.length),
-          dom.span({ style: { paddingLeft: 10 }}, 'Data points: ', (props.assessments || []).length)));
+          dom.span({ style: { paddingLeft: 10 }}, '# Data points: ', (props.assessments || []).length)));
```
Comment (kevinrobinson) [inline app/assets/javascripts/components/star_charts_page.js]: Using the symbol might be cryptic, how about "Number of data points" if that fits?

File: app/assets/javascripts/school_overview/school_overview_page.js [hunk 4]
```diff
@@ -28,13 +29,15 @@
...
+    //  $(document).on('keydown', this.onKeyDown);
+    //},
 
-    componentWillUnmount: function() {
-      $(document).off('keydown', this.onKeyDown);
-    },
+    // Moved to slice_buttons.js
+    //componentWillUnmount: function() {
+    //  $(document).off('keydown', this.onKeyDown);
+    //},
```
Comment (kevinrobinson) [inline app/assets/javascripts/school_overview/school_overview_page.js]: We can remove these comments now, right?

File: Gemfile.lock [hunk 5]
```diff
@@ -328,5 +328,8 @@ DEPENDENCIES
   wicked_pdf
   wkhtmltopdf-binary
 
+RUBY VERSION
+   ruby 2.3.0p0
+
 BUNDLED WITH
-   1.11.2
+   1.12.4
```
Comment (kevinrobinson) [inline Gemfile.lock]: Is this intentional?  Unless it's necessary let's split out upgrades to dependencies into a separate PR.

**Comment by kevinrobinson** [comment]
@jonchar Awesome!  Could you also drop in screenshots of what this looks like?  The comments are mostly about cleanup so should be straightforward to finish off this great improvement!

**Comment by jonchar** [comment]
Thanks for the comments @kevinrobinson ! I think I've addressed them all. Here is a screenshot of how the `SliceButtons` look on the STAR Reading page.

slicebuttons_on_star_page

**Comment by kevinrobinson** [comment]
@jonchar awesome! 😄
